### PR TITLE
First pass at the serial property

### PR DESCRIPTION
**Pull request by wtoorop**

### CODE REVIEW

File: draft-ietf-dnsop-dns-catalog-zones.md [hunk 1]
```diff
@@ -210,6 +210,89 @@ RECOMMENDED.
...
+for a member zone in the catalog, implementations of catalog zones MAY assume
+this number to be the current serial number in the SOA record of the most
+recent version of the member zone.
+
+Nameservers that are secondary for that member zone, MAY compare the `serial`
+property with the SOA serial since the last time the zone was fetched. When the
+`serial` property is larger, the secondary MAY initiate a zone tranfer
+immediately without doing an SOA query first. The transfer MUST be aborted
+immediately when the serial number of the SOA resource record does not match
+the `serial` property. In that case the zone transfer should be retried after
```
[the remark this answers is not in the document]
Comment (wtoorop) [reply draft-ietf-dnsop-dns-catalog-zones.md]: Absolutely! Reworded to:
```
The transfer MUST be aborted
immediately when the serial number of the SOA resource record in the transfer
is not larger than the SOA serial of the zone currently being served. In that
case the zone transfer should be retried after the time given in the retry
field of the SOA record of the member zone, or earlier if a new SOA serial
number is learned via an updated `serial` property, or via NOTIFY [@!RFC1996].
```

File: draft-ietf-dnsop-dns-catalog-zones.md [hunk 2]
```diff
@@ -210,6 +210,89 @@ RECOMMENDED.
...
+the time given in the retry field of the SOA record of the member zone, or
+earlier if a new SOA serial number is learned (via an updated `serial` property
+of the member zone, or via NOTIFY [@!RFC1996]).
+
```
[the remark this answers is not in the document]
Comment (wtoorop) [reply draft-ietf-dnsop-dns-catalog-zones.md:241]: Yes, I added a paragraph towards the end to state this:
```
Primary nameservers MAY be configured to omit sending DNS NOTIFY messages to
secondary nameservers which are known to process the `serial` property of the
member zones in that catalog. However they MAY also combine signalling of zone
changes with the `serial` property of a member zone, as well as sending DNS
NOTIFY messages, to anticipate slow updates of the catalog zone (due to packet
loss or other reasons) and to cater for secondaries that do not process the
`serial` property.
```

File: draft-ietf-dnsop-dns-catalog-zones.md [hunk 3]
```diff
@@ -210,6 +210,89 @@ RECOMMENDED.
 Each RRSet of catalog zone, with the exception of the zone apex, SHOULD consist of just one RR. It's acceptable to generate owner names with the help of a
 sufficiently strong hash function, with small probablity that unrelated records fall within the same RRSet.
 
+# The Serial Property
+
+The current default mechanism for prompting notifications of zone changes from
+a primary nameserver to the secondaries via DNS NOTIFY [@!RFC1996], can be
+unreliable due to lost UDP packets used to send the NOTIFY, or secondary
```
[the remark this answers is not in the document]
Comment (wtoorop) [reply draft-ietf-dnsop-dns-catalog-zones.md]: Done

File: draft-ietf-dnsop-dns-catalog-zones.md [hunk 4]
```diff
@@ -210,6 +210,89 @@ RECOMMENDED.
...
+nameservers temporarily not being reachable. In such cases the secondary might
+pickup the change only after the refresh timer runs out, which might be long
+and out of the control of the DNS operator.  Alternatively updates of zones MAY
```
[the remark this answers is not in the document]
Comment (wtoorop) [reply draft-ietf-dnsop-dns-catalog-zones.md]: Good point! Taken along with slightly different words. Thanks!

File: draft-ietf-dnsop-dns-catalog-zones.md [hunk 5]
```diff
@@ -210,6 +210,142 @@ RECOMMENDED.
...
+
+## SERIAL Presentation Format {#serialrrpf}
+
+The presentation format of the RDATA portion is as follows:
+
+The Serial fields is represented as an unsigned decimal integer.
+
+## SERIAL RR Usage - option 1 {#serialrr1}
+
+The `serial` property of a member zone is provided by an SERIAL RRset with a
```
Comment (Habbie) [inline draft-ietf-dnsop-dns-catalog-zones.md:292]:
```suggestion
The `serial` property of a member zone is provided by a SERIAL RRset with a
```

File: draft-ietf-dnsop-dns-catalog-zones.md [hunk 6]
```diff
@@ -210,6 +210,142 @@ RECOMMENDED.
...
+<m-unique-1>.zones.$CATZ 0 IN SERIAL 2020111712
+<m-unique-2>.zones.$CATZ 0 IN PTR    example.net.
+<m-unique-2>.zones.$CATZ 0 IN SERIAL 2020111709
+<m-unique-3>.zones.$CATZ 0 IN PTR    example.org.
+<m-unique-3>.zones.$CATZ 0 IN SERIAL 2020112405
+~~~
+
+## Serial property as TXT RR - option 3 {#serialrr3}
+
+The `serial` property of a member zone is provided by an TXT RRset with a
```
Comment (Habbie) [inline draft-ietf-dnsop-dns-catalog-zones.md:328]:
```suggestion
The `serial` property of a member zone is provided by a TXT RRset with a
```

File: draft-ietf-dnsop-dns-catalog-zones.md [hunk 7]
```diff
@@ -210,6 +210,142 @@ RECOMMENDED.
...
+## SERIAL RR Usage - option 1 {#serialrr1}
+
+The `serial` property of a member zone is provided by an SERIAL RRset with a
+single SERIAL RR named `serial.<m-unique-N>.zones.$CATZ`.
+
+For example, if a catalog zone lists three zones "example.com.", "example.net."
+and "example.org.", and a `serial` property is provided for each of them, the
+RRs would appear as follows:
+
+~~~ asci-art
```
Comment (Habbie) [inline draft-ietf-dnsop-dns-catalog-zones.md:299]: Should this say `ascii-art`?

**Review by Habbie — APPROVED** [review]

File: draft-ietf-dnsop-dns-catalog-zones.md (hunk 7 above)
Comment (wtoorop) [reply draft-ietf-dnsop-dns-catalog-zones.md]: No better to do just ```

File: draft-ietf-dnsop-dns-catalog-zones.md [hunk 8]
```diff
@@ -210,6 +210,142 @@ RECOMMENDED.
...
+for a member zone, implementations of catalog zones MAY assume this number to
+be the current serial number in the SOA record of the most recent version of
+the member zone.
+
+Nameservers that are secondary for that member zone, MAY compare the `serial`
+property with the SOA serial since the last time the zone was fetched. When the
+`serial` property is larger, the secondary MAY initiate a zone tranfer
+immediately without doing a SOA query first. The transfer MUST be aborted
+immediately when the serial number of the SOA resource record in the transfer
+is not larger than the SOA serial of the zone currently being served. In that
```
[the remark this answers is not in the document]
Comment (wtoorop) [reply draft-ietf-dnsop-dns-catalog-zones.md:237]: How? And does it matter? Sometimes IXFR is not an option and you are forced to do AXFR.

**Pull request merged**